extern/storage-sealing/fsm.go

```diff
@@ -136,6 +137,7 @@ var fsmPlanners = map[SectorState]func(events []statemachine.Event, state *Secto
 
 	FinalizeSector: planOne(
 		on(SectorFinalized{}, Proving),
+		on(SectorFinalizedAvailable{}, Available),
```

Comment:
You can't really distinguish CC sectors already in the proving state from CC sectors entering the proving state (that's because on restart we run all state handlers for current states of all sectors). If we were to do that without tracking existing CC sectors, all CC sectors would get converted to `Available`, which I'm quite sure is not what SPs will want.

Other things we could do instead of this event:
* Additional state deciding this
  * Now we need 2 events, and an additional state
* Add a boolean to `SectorInfo` to track which CC sectors are Proving already, and shouldn't be auto-converted to Available
  * Pretty sure this still requires an additional event to update this variable. Also adds more bits of private per-sector state, which ideally wouldn't exist

Reply:
Thanks makes sense.  That run all state handlers on restart fact is pretty important.